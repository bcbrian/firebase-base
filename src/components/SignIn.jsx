import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import TextField from '@material-ui/core/TextField';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

export default class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    };
  }
  handleEmailOnChange = ({ target }) => {
    this.setState(() => ({
      email: target.value
    }));
  };

  handlePasswordOnChange = ({ target }) => {
    this.setState(() => ({
      password: target.value
    }));
  };

  handleSubmit = e => {
    e.preventDefault();
    const { email, password } = this.state;
    this.props.onSignIn({ email, password });
  };

  render() {
    const { open, toggle } = this.props;
    const { email, password } = this.state;
    return (
      <Dialog open={open} onClose={toggle} aria-labelledby="form-dialog-title">
        <div>
          <DialogTitle id="form-dialog-title">Sign In</DialogTitle>
          <form onSubmit={this.handleSubmit}>
            <DialogContent>
              <DialogContentText>
                Lets get you signed in so you can know whats left.
              </DialogContentText>
              <TextField
                onChange={this.handleEmailOnChange}
                value={email}
                type="text"
                name="email"
                label="Email"
                variant="outlined"
              />
              <TextField
                onChange={this.handlePasswordOnChange}
                value={password}
                type="password"
                name="pasword"
                label="Password"
                variant="outlined"
              />
            </DialogContent>
            <DialogActions>
              <Button onClick={toggle} color="secondary">
                Cancel
              </Button>
              <Button variant="contained" color="primary" type="submit">
                Sign In
              </Button>
            </DialogActions>
          </form>
        </div>
      </Dialog>
    );
  }
}

SignIn.propTypes = {
  onSignIn: PropTypes.func.isRequired
};
