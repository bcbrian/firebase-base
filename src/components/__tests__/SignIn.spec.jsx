import React from 'react';
import renderer from 'react-test-renderer';
import SignIn from '../SignIn';

describe('SignIn Component', () => {
  it('should render sign up', () => {
    const onSignIn = jest.fn();
    const SignInComp = renderer.create(<SignIn onSignIn={onSignIn} />);
    expect(SignInComp.toJSON()).toMatchSnapshot();
  });

  it('should call the onSignIn prop function with {email, passord}', () => {
    const email = 'test@test.com';
    const password = '1234abcd';
    const onSignIn = jest.fn();
    const SignInComp = renderer.create(<SignIn onSignIn={onSignIn} />);
    SignInComp.getInstance().setState({ email, password });
    SignInComp.getInstance().handleSubmit({ preventDefault: jest.fn() });
    expect(onSignIn).toHaveBeenCalledWith({ email, password });
    expect(SignInComp.toJSON()).toMatchSnapshot();
  });
});
