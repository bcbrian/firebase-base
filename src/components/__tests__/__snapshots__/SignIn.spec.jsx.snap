// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Component should call the onSignIn prop function with {email, passord} 1`] = `
<div>
  <h3
    className="ui dividing header"
  >
    Sign In
  </h3>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className="ui left icon input"
      >
        <i
          aria-hidden="true"
          className="at icon"
        />
        <input
          disabled={undefined}
          name="email"
          onChange={[Function]}
          placeholder="email"
          tabIndex={undefined}
          type="text"
          value="test@test.com"
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui left icon input"
      >
        <i
          aria-hidden="true"
          className="asterisk icon"
        />
        <input
          disabled={undefined}
          name="pasword"
          onChange={[Function]}
          placeholder="password"
          tabIndex={undefined}
          type="password"
          value="1234abcd"
        />
      </div>
    </div>
    <button
      className="ui fluid primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="submit"
    >
      Sign In
    </button>
  </form>
</div>
`;

exports[`SignIn Component should render sign up 1`] = `
<div>
  <h3
    className="ui dividing header"
  >
    Sign In
  </h3>
  <form
    action={undefined}
    className="ui form"
    onSubmit={[Function]}
  >
    <div
      className="field"
    >
      <div
        className="ui left icon input"
      >
        <i
          aria-hidden="true"
          className="at icon"
        />
        <input
          disabled={undefined}
          name="email"
          onChange={[Function]}
          placeholder="email"
          tabIndex={undefined}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="field"
    >
      <div
        className="ui left icon input"
      >
        <i
          aria-hidden="true"
          className="asterisk icon"
        />
        <input
          disabled={undefined}
          name="pasword"
          onChange={[Function]}
          placeholder="password"
          tabIndex={undefined}
          type="password"
          value=""
        />
      </div>
    </div>
    <button
      className="ui fluid primary button"
      disabled={undefined}
      onClick={[Function]}
      role="button"
      tabIndex={undefined}
      type="submit"
    >
      Sign In
    </button>
  </form>
</div>
`;
